plugins/polio/api.py

```diff
@@ -2082,6 +2091,230 @@ def get_queryset(self):
+
+    if "date" not in ordering:
+        item = item.get(ordering)
+        return item if item is not None else float("inf")
+
+
+@swagger_auto_schema(tags=["vaccineauthorizations"])
+class VaccineAuthorizationViewSet(ModelViewSet):
+    """
+    Vaccine Authorizations API
+    list: /api/polio/vaccintauthorizations
+    action: /api/polio/get_most_recent_authorizations
+    """
+
+    permission_classes = [HassVaccineAuthorizationsPermissions]
+    filter_backends = [filters.OrderingFilter, DjangoFilterBackend, DeletionFilterBackend]
+    results_key = "results"
+    remove_results_key_if_paginated = True
+    serializer_class = VaccineAuthorizationSerializer
+    pagination_class = Paginator
+    ordering_fields = ["status", "current_expiration_date", "next_expiration_date", "expiration_date", "quantity"]
+
+    def get_queryset(self):
+        user = self.request.user
+        user_access_ou = OrgUnit.objects.filter_for_user_and_app_id(user, None)
+        user_access_ou = user_access_ou.filter(org_unit_type__name="COUNTRY")
+        country_id = self.request.query_params.get("country", None)
+        queryset = VaccineAuthorization.objects.filter(account=user.iaso_profile.account, country__in=user_access_ou)
+        block_country = self.request.query_params.get("block_country", None)
+        search = self.request.query_params.get("search", None)
+        auth_status = self.request.query_params.get("auth_status", None)
+
+        if country_id:
+            queryset = queryset.filter(country__pk=country_id)
+        if block_country:
+            block_country = block_country.split(",")
+            block_country = Group.objects.filter(pk__in=block_country)
+            org_units = [
+                ou_queryset
+                for ou_queryset in [
+                    ou_group_queryset for ou_group_queryset in [country.org_units.all() for country in block_country]
+                ]
+            ]
+            ou_pk_list = []
+            for ou_q in org_units:
+                for ou in ou_q:
+                    ou_pk_list.append(ou.pk)
+            queryset = queryset.filter(country__pk__in=ou_pk_list)
+        if search:
+            queryset = queryset.filter(country__name__icontains=search)
+        if auth_status:
+            auth_status = auth_status.split(",")
+            queryset = queryset.filter(status__in=auth_status)
+
+        return queryset
+
+    def create(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        country = self.request.data.get("country")
+        if not self.request.user.is_superuser:
+            if country not in [
+                str(ou.id) for ou in OrgUnit.objects.filter_for_user_and_app_id(self.request.user, None)
+            ]:
+                raise serializers.ValidationError({"Error": "You don't have access to this org unit."})
+
+        return super().create(request)
+
+    @action(detail=False, methods=["POST", "GET", "PUT", "DELETE"])
```

Comment:
Not sure why we are allowing all theses methods ? Shouldn't it be only GET or I'm missing something ?